arch/cortex-m0/src/lib.rs

```diff
@@ -77,7 +73,7 @@ pub unsafe extern "C" fn switch_to_user(mut user_stack: *const u8,
 
     "
     : "=r"(user_stack)
-    : "r"(user_stack), "r"(process_got), "r"(process_regs)
+    : "r"(user_stack), "r"(process_regs)
```

Comment:
Oh! Duh, `r0` is handled by the output `"=r"(user_stack)`, it shouldn't be a clobber. _However_, I do think that we should make this change:

```diff
     "
     : "=r"(user_stack)
-    : "r"(user_stack), "r"(process_got), "r"(process_regs)
+    : "r0"(user_stack), "r"(process_got), "r"(process_regs)
     : "r4","r5","r6","r7","r8","r9","r10","r11");
     user_stack as *mut u8
```

which indicates that the 0th input is overwritten by the 0th output

https://gcc.gnu.org/onlinedocs/gcc-6.3.0/gcc/Extended-Asm.html#InputOperands
> Warning: Do not modify the contents of input-only operands (except for inputs tied to outputs). The compiler assumes that on exit from the asm statement these operands contain the same values as they had before executing the statement. It is not possible to use clobbers to inform the compiler that the values in these inputs are changing.